prealloc.go

```diff
@@ -215,13 +226,29 @@ func (v *returnsVisitor) Visit(node ast.Node) ast.Visitor {
 					if !ok {
 						continue
 					}
-					if genD.Tok == token.VAR {
+					if genD.Tok == token.TYPE {
+						for _, spec := range genD.Specs {
+							tSpec, ok := spec.(*ast.TypeSpec)
+							if !ok {
+								continue
+							}
+
+							if _, ok := tSpec.Type.(*ast.ArrayType); ok {
+								arrayTypes = append(arrayTypes, tSpec.Name.Name)
+							}
+						}
+					} else if genD.Tok == token.VAR {
 						for _, spec := range genD.Specs {
 							vSpec, ok := spec.(*ast.ValueSpec)
 							if !ok {
 								continue
 							}
-							if /*arrayType*/ _, ok := vSpec.Type.(*ast.ArrayType); ok {
+							var ok2 bool
+							/*arrayType*/ _, ok1 := vSpec.Type.(*ast.ArrayType)
+							if val, ok := vSpec.Type.(*ast.Ident); ok {
```

Comment:
Instead of having `ok` and `ok1` here, could we do a [type switch](https://tour.golang.org/methods/16) on `vSpec.Type`?